core/src/main/java/io/questdb/griffin/engine/functions/catalogue/TypeCatalogueFunctionFactory.java

```diff
@@ -24,32 +24,15 @@
 
 package io.questdb.griffin.engine.functions.catalogue;
 
-import io.questdb.cairo.*;
+import io.questdb.cairo.CairoConfiguration;
 import io.questdb.cairo.sql.Function;
-import io.questdb.cairo.sql.RecordCursor;
-import io.questdb.cairo.sql.RecordMetadata;
 import io.questdb.griffin.FunctionFactory;
-import io.questdb.griffin.SqlExecutionContext;
-import io.questdb.griffin.engine.EmptyTableRecordCursor;
 import io.questdb.griffin.engine.functions.CursorFunction;
-import io.questdb.std.Misc;
+import io.questdb.griffin.engine.functions.GenericRecordCursorFactory;
 import io.questdb.std.ObjList;
-import io.questdb.std.str.Path;
 
 public class TypeCatalogueFunctionFactory implements FunctionFactory {
```

Comment:
check if this factory overrides `isCursor()`

isCursor must return `true` for all factories that return cursor instance